client/src/main/java/com/netflix/conductor/client/http/ClientBase.java

```diff
@@ -189,15 +197,53 @@ private Builder getWebResourceBuilder(URI URI, Object entity) {
         return client.resource(URI).type(MediaType.APPLICATION_JSON).entity(entity).accept(MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON);
     }
 
-    private void handleException(URI uri, RuntimeException e) {
-        if (e instanceof ClientHandlerException) {
-            logger.error("Unable to invoke Conductor API with uri: {}, failure to process request or response", uri, e);
-        } else if (e instanceof UniformInterfaceException) {
+    private void handleClientHandlerException(ClientHandlerException exception, URI uri){
+        String errorMessage = String.format("Unable to invoke Conductor API with uri: %s, failure to process request or response", uri);
+        logger.error(errorMessage);
+        throw new ConductorClientException(errorMessage, exception);
+    }
+
+    private void handleRuntimeException(RuntimeException exception, URI uri) {
+        String errorMessage = String.format("Unable to invoke Conductor API with uri: %s, runtime exception occurred", uri);
+        logger.error(errorMessage);
+        throw new ConductorClientException(errorMessage, exception);
+    }
+
+    private void handleUniformInterfaceException(ClientResponse clientResponse, URI uri) {
```

Comment:
should take UniformInterfaceException as argument instead of clientResponse, you can get clientResponse from UniformInterfaceException.